netobserv/v1alpha1/flowcollector_types.go

```diff
@@ -0,0 +1,436 @@
+	// +optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,8,opt,name=resources"`
+
+	//+kubebuilder:default:=true
+	// EnableKubeProbes is a flag to enable or disable Kubernetes liveness/readiness probes
+	EnableKubeProbes bool `json:"enableKubeProbes,omitempty"`
+}
+
+type FlowCollectorHPA struct {
+	// minReplicas is the lower limit for the number of replicas to which the autoscaler
+	// can scale down.  It defaults to 1 pod.  minReplicas is allowed to be 0 if the
+	// alpha feature gate HPAScaleToZero is enabled and at least one Object or External
+	// metric is configured.  Scaling is active as long as at least one metric value is
+	// available.
+	// +optional
+	MinReplicas *int32 `json:"minReplicas,omitempty" protobuf:"varint,2,opt,name=minReplicas"`
+	// upper limit for the number of pods that can be set by the autoscaler; cannot be smaller than MinReplicas.
+	MaxReplicas int32 `json:"maxReplicas" protobuf:"varint,3,opt,name=maxReplicas"`
+	// Metrics used by the pod autoscaler
+	// +optional
+	Metrics []ascv2.MetricSpec `json:"metrics"`
+}
+
+// FlowCollectorLoki defines the desired state for FlowCollector's Loki client
+type FlowCollectorLoki struct {
+	//+kubebuilder:default:="http://loki:3100/"
+	// URL is the address of an existing Loki service to push the flows to.
+	URL string `json:"url,omitempty"`
+
+	//+kubebuilder:validation:optional
+	// QuerierURL specifies the address of the Loki querier service, in case it is different from the
+	// Loki ingester URL. If empty, the URL value will be used (assuming that the Loki ingester
+	// and querier are int he same host).
+	QuerierURL string `json:"querierUrl,omitempty"`
+
+	//+kubebuilder:default:="1s"
+	// BatchWait is max time to wait before sending a batch
+	BatchWait metav1.Duration `json:"batchWait,omitempty"`
+
+	//+kubebuilder:validation:Minimum=1
+	//+kubebuilder:default:=102400
+	// BatchSize is max batch size (in bytes) of logs to accumulate before sending
+	BatchSize int64 `json:"batchSize,omitempty"`
+
+	//+kubebuilder:default:="10s"
+	// Timeout is the maximum time connection / request limit
+	// A Timeout of zero means no timeout.
+	Timeout metav1.Duration `json:"timeout,omitempty"`
+
+	//+kubebuilder:default:="1s"
+	// MinBackoff is the initial backoff time for client connection between retries
+	MinBackoff metav1.Duration `json:"minBackoff,omitempty"`
+
+	//+kubebuilder:default:="300s"
+	// MaxBackoff is the maximum backoff time for client connection between retries
+	MaxBackoff metav1.Duration `json:"maxBackoff,omitempty"`
+
+	//+kubebuilder:validation:Minimum=0
+	//+kubebuilder:default:=10
+	// MaxRetries is the maximum number of retries for client connections
+	MaxRetries int32 `json:"maxRetries,omitempty"`
+
+	//+kubebuilder:default:={"app":"netobserv-flowcollector"}
+	// StaticLabels is a map of common labels to set on each flow
+	StaticLabels map[string]string `json:"staticLabels,omitempty"`
```

Comment:
This looks maybe like the objectmeta style objects, maybe you want to allow others like annotations as well? In which case, embedding this under a `metadata` would make it look more kube like and would help the user understand through familiarity what is happening here

Reply:
`labels` here doesn't refer to kube-like labels, but to prometheus-like (or loki) labels, which are not only descriptive but are critical for the data store (they are actually indexes). So I don't think we _want_ here to make them look kube-like.